frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -303,16 +317,18 @@ export const transformTopologyData = (
   cheURL?: string,
   utils?: Function[],
 ): TopologyDataModel => {
-  const installedOperators = _.get(resources, 'clusterServiceVersion.data');
-  const operatorBackedServiceKinds = [];
+  const installedOperators = _.get(resources, 'clusterServiceVersions.data');
+  let operatorBackedServiceKindMap: OperatorBackedServiceKindMap;
   const serviceBindingRequests = _.get(resources, 'serviceBindingRequests.data');
-
   if (installedOperators) {
-    _.forEach(installedOperators, (op) => {
-      _.get(op, 'spec.customresourcedefinitions.owned').map((service) =>
-        operatorBackedServiceKinds.push(service.kind),
-      );
-    });
+    operatorBackedServiceKindMap = installedOperators.reduce((kindMap, csv) => {
+      _.get(csv, 'spec.customresourcedefinitions.owned').forEach((crd) => {
```

Comment:
Is there any concern that an installed operator would not have CRDs? Is this somehow guaranteed by the spec never to be missing?

Reply:
Based on the Operator SDK - https://github.com/operator-framework/operator-sdk#workflow, there can be operators which do not provide any CRDs.

Reply:
I guess my question is more about your `_.get` doesn't handle a fallback if it is missing.

Reply:
But, I am yet to find an operator from OperatorHub which is not providing any APIs.

Reply:
I suppose I am being overly cautious... defaulting it to an array saves us from a NPE if it happens. 🤷‍♂ It probably isn't needed one way or the other though. Just a recommendation for being "defensive" in programming.

Reply:
Added default array. It's good to be cautious. :)